chain/client/src/stateless_validation/state_witness_producer.rs

```diff
@@ -48,23 +48,22 @@ impl Client {
             transactions_storage_proof,
         )?;
 
-        // TODO(#10508): Remove this block once we have better ways to handle chunk state witness and
-        // chunk endorsement related network messages.
         if let Some(my_signer) = self.validator_signer.clone() {
-            let validator_id = my_signer.validator_id();
-            if chunk_validators.contains(validator_id) {
-                // Endorse the chunk immediately, bypassing sending state witness
-                // to ourselves, because network can't send messages to ourselves.
-                // Also useful in tests where we don't have a good way to handle
-                // network messages and there's only a single client.
+            // Remove ourselves from the list of chunk validators. Network can't send messages to ourselves.
+            chunk_validators.retain(|validator| {
```

Comment:
It's a bit strange to do the bypass inside the retain method closure tbh. I would suggest doing retain and bypass separately.

Reply:
Yeah, makes sense...